src/lua/circllhist.lua

```diff
@@ -0,0 +1,565 @@
+function Histogram_C:funiter()
+  local param = { self = self, limit = self:bucket_count()-1 }
+  local state = 0
+  return fun.iter(histogram_C_funiter_next, param, state)
+    :filter(histogram_filter_count_zero)
+end
+
+function Histogram_C:bucket_idx(bucket_id)
+  local value =  ffi.new("double[1]")
+  local count =  ffi.new("uint64_t[1]")
+  libhist.hist_bucket_idx(self, bucket_id, value, count)
+  return tonumber(value[0]), tonumber(count[0])
+end
+
+function Histogram_C:merge(other_histogram)
+  for _, b, c in other_histogram:funiter() do
+    self:insert(b,c);
+  end
+end
+
+function Histogram_C:merge_c(other_histogram_c)
+  local other_histo_ptr = ffi.new("const histogram_t*[1]", other_histogram_c)
+  libhist.hist_accumulate(self, other_histo_ptr, 1);
+end
+
+--- insert data into the histogram.
+--@param value, this does *not* need to be a bucket boundary
+--@param count, how many samples of that value to insert
+function Histogram_C:insert(value, count)
+  if not value then return end
+  libhist.hist_insert(self, value, count or 1)
+end
+
+function Histogram_C:mean()
+  return libhist.hist_approx_mean(self)
+end
+
+function Histogram_C:sum()
+  return libhist.hist_approx_sum(self)
+end
+
+function Histogram_C:_quantiles(quantiles)
+  -- TODO: Fix empty histogram handling. Currently this returns 0 instead of nil.
+  if self:isempty() then return {} end
+  local q_in = ffi.new("double[?]", #quantiles, quantiles)
+  local q_out = ffi.new("double[?]", #quantiles, 0/0)
+  local rc = libhist.hist_approx_quantile(self, q_in, #quantiles, q_out)
+  if rc > 0 then error("hist_approx_quantile failed with RC: " .. tostring(rc)) end
+  local out = {}
+  for i=1,#quantiles do
+    out[i] = q_out[i-1]
+  end
+  return out
+end
+
+function Histogram_C:serialize()
+  local buf_length = libhist.hist_serialize_b64_estimate(self)
+  local buf = ffi.gc(ffi.C.malloc(buf_length), ffi.C.free)
+  assert(buf ~= nil, "ffi.C.malloc("..tonumber(buf_length)..") returned nil (out of memory?!)")
+
+  local string_length = libhist.hist_serialize_b64(self, buf, buf_length)
+  return ffi.string(buf, string_length)
+end
+
+function Histogram_C:compress_mbe(mbe)
+  local c_hist_mbe = libhist.hist_compress_mbe(self, mbe)
+  local free_when_done = true
+  return Histogram_C:from_c_struct(c_hist_mbe, free_when_done)
+end
+
+local ffi_bucket_str_buffer = ffi.new("char[128]")
+local function ffi_hist_bucket_to_string(ffi_bucket, format)
+  if format == "double" then
+    local bucket_str_len = libhist.hist_bucket_to_string(ffi_bucket, ffi_bucket_str_buffer)
+    return ffi.string(ffi_bucket_str_buffer, bucket_str_len)
+  elseif format == "hex" then
+    return '0x' .. bit.tohex(bit.bor(ffi_bucket.exp, bit.lshift(ffi_bucket.val,8)), 4)
+  else
+    return string.format(format, libhist.hist_bucket_to_double(ffi_bucket))
+  end
+end
+-- fast, standartized JSON encoder
+-- This function will be used by dkjson to serialize Histogam_C objects
+function Histogram_C:__tojson(state)
```

Comment:
Just call this `:tojson()` like we have `:tostring()` or `:totable()`. Users might not care about dkjson.